### PR TITLE
Cache Conda env

### PR DESCRIPTION
**Pull request by jonashaag**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: .github/actions/setup-conda/action.yml [hunk 1]
```diff
@@ -24,14 +24,13 @@ runs:
...
-      uses: conda-incubator/setup-miniconda@v2.1.1
+      uses: mamba-org/provision-with-micromamba@v12
       with:
         environment-file: ${{ inputs.environment-file }}
-        activate-environment: ${{ inputs.environment-name }}
-        python-version: ${{ inputs.python-version }}
-        channel-priority: ${{ runner.os == 'macOS' && 'flexible' || 'strict' }}
+        environment-name: ${{ inputs.environment-name }}
+        extra-specs: |
+          python =${{ inputs.python-version }}
```
Comment (mroeschke) [inline .github/actions/setup-conda/action.yml:32]: So this is to specifically support the sdist workflow? (might be good to add a comment to that affect if so)

**Review by mroeschke — COMMENTED** [review]
1. Can `cache-downloads: true` be set?
2. Could you give an example before/after improvement in the setup?

File: .github/actions/setup-conda/action.yml (hunk 1 above)
Comment (jonashaag) [reply .github/actions/setup-conda/action.yml]: Yes, actually come to think of it I think it's better to simply pass through `extra-specs`. It's more flexible.

**Comment by jonashaag** [comment]
> 1. Can `cache-downloads: true` be set?

Yes we can do it. I'm not sure it's worth it given that the total amount of data that can be cached per repository is 10 GB (https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) and some of the envs are 0.5 GB large.

**Comment by mroeschke** [comment]
> > 1. Can `cache-downloads: true` be set?
> 
> Yes we can do it. I'm not sure it's worth it given that the total amount of data that can be cached per repository is 10 GB (https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) and some of the envs are 0.5 GB large.

I see. I would still be partial in trying it as I've noticed we sometimes still hit connection errors from conda although I extended the timeouts in the `condarc.yml`. If environments are evicted from the cache during every build due to the size then yeah I can see it not being worth it

File: .github/workflows/sdist.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ on:
 
 jobs:
   build:
-    if: ${{ github.event.label.name == 'Build' || contains(github.event.pull_request.labels.*.name, 'Build') || github.event_name == 'push'}}
```
Comment (mroeschke) [inline .github/workflows/sdist.yml:18]: Looks like this should be reverted now

**Review by mroeschke — APPROVED** [review]

**Pull request merged**